src/element_processing/water_areas.rs

```diff
@@ -36,70 +36,32 @@ pub fn generate_water_areas(editor: &mut WorldEditor, element: &ProcessedRelatio
         }
     }
 
-    // Process each outer polygon individually
-    for (i, outer_nodes) in outers.iter().enumerate() {
-        let mut individual_outers = vec![outer_nodes.clone()];
-
-        merge_loopy_loops(&mut individual_outers);
-        if !verify_loopy_loops(&individual_outers) {
-            println!(
-                "Skipping invalid outer polygon {} for relation {}",
-                i + 1,
-                element.id
-            );
-            continue; // Skip this outer if it's not valid
-        }
+    merge_loopy_loops(&mut outers);
+    if !verify_loopy_loops(&outers) {
+        println!("Skipping relation {} due to invalid polygon", element.id);
+        return;
+    }
 
-        merge_loopy_loops(&mut inners);
-        if !verify_loopy_loops(&inners) {
-            // If inners are invalid, process outer without inners
```

Comment:
This also didn't seem like a great fallback to me. If the inner loop is invalid we probably don't want to just fill it in!